### DISCUSSION TITLE
Do alerts need to be filtered?

### DISCUSSION THREAD

**Discussion by jhemak** [weather-gov/api#764] [Q&A]
Hello!  When I query my location using api.weather.gov I see two heat advisories.  Everything seems the same except for the value of "expires" .  When I query the same location using just about any weather site -- weather.com, weather.gov, wunderground, etc. I see only one heat advisory which is the one more recently issued.  Obviously all these sites are somehow filtering out the second one, but I do not see anything in the API response that would tell me to use one and ignore the other.  Any idea what logic others are using here?

**Comment by sullynole** [comment]
What is the request you are using?

**Comment by jhemak** [comment]
https://api.weather.gov/alerts/active.atom?zone=OHZ077%2COHC061

**Reply by sullynole** [reply]
I don't see any heat advisories, but there are two Air Quality alerts that were issued at different times. Unfortunately since those products are relayed from external agencies they often show up duplicated that way since they are not coded from the source in a way that makes it clear to dedup.

**Reply by jhemak** [reply]
I’m sorry, you are right. I made a mistake when I said heat vs air quality. Nonetheless, there must be some logic that weather.gov and the other user-facing sites are implementing - rather consistently - in order to remove the duplicates.  Thanks for the insight though!

**Reply by sullynole** [reply]
Each agency's air quality alert criteria may be slightly different, but using the last issued in this case may work.  Check on the issuance criteria to see if that makes sense.

**Reply by scadergit** [reply]
If you're curious, weather.gov and the API have different data flows. The office generates an alert as a [text product](https://forecast.weather.gov/product.php?site=CRH&issuedby=ILN&product=AQA&format=CI&version=1&glossary=0). On one path, those are converted to CAP and ultimately feed the API. On another path, which is slowly geting replaced to the use API, the original text products are parsed and inserted into a database which utlimately feed weather.gov (which has no concept of CAP). The text product parsing for weather.gov has a long history of modifications to solve issues, including catching oddities in how text products are issued by the offices. 

As for weather.com--no idea, they probably have their own method.

You might also reach out to [the forecast office](mailto:iln.webmaster@noaa.gov) to ask why there were two identifical text products. I would think the second should have been an extension of the first, in which the API would only display the second.

**Reply by sullynole** [reply]
There is no VTEC in AQAs so that is the issue for CAP alert tracking. Agree it would be a good idea to reach out to the local office to find out if you can always use the latest one.

**Reply by jhemak** [reply]
Thanks, guys!  This is very helpful!
